### ISSUE TITLE
TRAPI 1.4 validation error: gain / loss of function qualifiers for Gene -> Disease

### ISSUE THREAD

**Issue by colleenXu** [biothings/biothings_explorer#699]
I was doing a [quick performance review of the new BTE "treats" templates](https://github.com/biothings/bte_trapi_query_graph_handler/pull/167#issuecomment-1681043185), and I used [my notebook](https://github.com/colleenXu/RegistryMetadataDev/blob/main/ValidatingBTE_Response.ipynb) to run TRAPI validation on some responses (attached below):

* Timothy syndrome `MONDO:0010979`: [timothy-syndrome.txt](https://github.com/biothings/biothings_explorer/files/12365948/timothy-syndrome.txt)
* nephrotic syndrome `MONDO:0005377`: [nephrotic.txt](https://github.com/biothings/biothings_explorer/files/12365952/nephrotic.txt)
* Familial dysautonomia / Riley-Day syndrome `MONDO:0009131`: [riley-day.txt](https://github.com/biothings/biothings_explorer/files/12365957/riley-day.txt)
* Huntington's disease `MONDO:0007739`: [huntington.txt](https://github.com/biothings/biothings_explorer/files/12365958/huntington.txt)

For all 4 of these responses, the reasoner-validator raises the error: [error.knowledge_graph.edge.qualifiers.qualifier.value.unresolved](https://ncatstranslator.github.io/reasoner-validator/validation_codes_dictionary.html#error-knowledge-graph-edge-qualifiers-qualifier-value-unresolved). Note that "errors" are "orange" in the ARAX UI, vs the red for "critical" errors. 

This occurs on KG Edges from Text-Mining Targeted like these:
* (subject_form_or_variant_qualifier: loss_of_function_variant_form) Gene -> affects -> Disease
* (subject_form_or_variant_qualifier: gain_of_function_variant_form) Gene -> affects -> Disease

Example error code and object from my notebook (python):
```
error.knowledge_graph.edge.qualifiers.qualifier.value.unresolved

{'infores:text-mining-provider-targeted -> infores:biothings-explorer': {'gain_of_function_variant_form': [{'edge_id': 'NCBIGene:775[biolink:Gene]--biolink:affects->MONDO:0010979[biolink:Disease]',
    'qualifier_type_id': 'biolink:subject_form_or_variant_qualifier'},
   {'edge_id': 'NCBIGene:6330[biolink:Gene]--biolink:affects->MONDO:0010979[biolink:Disease]',
    'qualifier_type_id': 'biolink:subject_form_or_variant_qualifier'},
   {'edge_id': 'NCBIGene:857[biolink:Gene]--biolink:affects->MONDO:0010979[biolink:Disease]',
    'qualifier_type_id': 'biolink:subject_form_or_variant_qualifier'}]}}
```

---

As far as I can tell, this error has to do with the reasoner-validator and data-modeling (biolink-model + text-mining KP). 

Reasoner-validator may depend on biolink-model having an association object that matches these Edges AND sets the values for its "subject form or variant qualifier". 
* There is a very similar association object (uses `contributes_to` as the predicate), but [it doesn't set values for the""subject form or variant qualifier"](https://github.com/biolink/biolink-model/blob/cfd18406cb7b6d2d2e04935b159c03cf5fefdaeb/biolink-model.yaml#L10272). When I try to match the KG edges to this association (by replacing the `affects` predicate with `contributes_to`), the error still occurs.
* In comparison: the two chem-gene association objects set the "subject form or variant qualifier" range [here](https://github.com/biolink/biolink-model/blob/cfd18406cb7b6d2d2e04935b159c03cf5fefdaeb/biolink-model.yaml#L9388) and [here](https://github.com/biolink/biolink-model/blob/cfd18406cb7b6d2d2e04935b159c03cf5fefdaeb/biolink-model.yaml#L9427).  
* This is reminding me of a previous issue regarding the values qualifiers can have https://github.com/biolink/biolink-model/pull/1362

**Comment by colleenXu** [comment]
@tokebe @andrewsu

These edges are occurring because of the templates intended to [make the directionality of m4-Chem-Gene-DoP.json more specific](https://github.com/biothings/biothings_explorer/issues/461#issuecomment-1568934102). 

* see templateGroups.json changes in [PR](https://github.com/biothings/bte_trapi_query_graph_handler/pull/167/files)
* [Chem-increaseGeneWithFunctionLossIn-Disease](https://github.com/biothings/bte_trapi_query_graph_handler/blob/1478f32955143f5f94b33e8b3779233a4d02fabe/data/templates/Drug-treats-Disease/Chem-increaseGeneWithFunctionLossIn-Disease.json)
* [Chem-decreaseGeneWithFunctionGainIn-Disease.](https://github.com/biothings/bte_trapi_query_graph_handler/blob/1478f32955143f5f94b33e8b3779233a4d02fabe/data/templates/Drug-treats-Disease/Chem-decreaseGeneWithFunctionGainIn-Disease.json)

What do you think we should do? 
* I'd prefer to keep these templates, since users have been asking for info like this for a long time. However, I'm not sure if this TRAPI validation error is a hint of any issues downstream...
* I think this should be brought up with Text-Mining Provider and that they should take the lead on data-modeling / reasoner-validator discussions.

**Comment by andrewsu** [comment]
Based on the discussion in [this slack thread](https://ncatstranslator.slack.com/archives/C0156TX0VB2/p1692289434312119), we should undo our roll out of templates that take advantage of "directionality" in Gene/Variant -> Disease edges.  Unfortunately, it's just not ready from a data modeling perspective...

**Comment by colleenXu** [comment]
We decided to remove the two templates with "more specific directionality with qualifiers" from the "treats" template group, because of TRAPI validation / data-modeling concerns. This leaves us w/ 3 templates. 

I confirmed that removal of the templates addresses this issue. [When all 4 of the example diseases were reran](https://github.com/biothings/bte_trapi_query_graph_handler/pull/169#issuecomment-1682912267), their responses didn't trigger any validation errors. 

Example: rerun timothy syndrome response: [new_timothy.txt](https://github.com/biothings/biothings_explorer/files/12373251/new_timothy.txt)

---

Done editing the linked comment. No TRAPI errors (critical or normal) encountered at all!
